sdk/src/wallet/operations/transaction/prepare_transaction.rs

```diff
@@ -40,15 +40,15 @@ where
             output.verify_storage_deposit(storage_score_params)?;
         }
 
-        let is_burn_present = options.as_ref().map(|options| options.burn.is_some()).unwrap_or(false);
+        // let is_burn_present = options.as_ref().map(|options| options.burn.is_some()).unwrap_or(false);
 
-        // Validate the number of outputs. The validation shouldn't be performed if [`Burn`] is present.
-        // The outputs will be generated by the input selection algorithm (ISA).
-        if !OUTPUT_COUNT_RANGE.contains(&(outputs.len() as u16)) && !is_burn_present {
-            return Err(crate::types::block::Error::InvalidOutputCount(
-                TryIntoBoundedU16Error::Truncated(outputs.len()),
-            ))?;
-        }
+        // // Validate the number of outputs. The validation shouldn't be performed if [`Burn`] is present.
+        // // The outputs will be generated by the input selection algorithm (ISA).
+        // if !OUTPUT_COUNT_RANGE.contains(&(outputs.len() as u16)) && !is_burn_present {
+        //     return Err(crate::types::block::Error::InvalidOutputCount(
+        //         TryIntoBoundedU16Error::Truncated(outputs.len()),
+        //     ))?;
+        // }
```

Comment:
This check is already at the start of ISA, should we remove it ?

Reply:
Guess it's fine to remove since most of the time less outputs should be provided and therefore the early return is not that important

Reply:
Removed